src/components/repositories/remote-form.tsx

```diff
@@ -733,6 +733,13 @@ export class RemoteForm extends React.Component<IProps, IState> {
       }
     }
 
+    if (
+      validateURLHelper(this.props.errorMessages['url'], remote.url)
```

Comment:
```suggestion
      validateURLHelper(null, remote.url)
```

Because `errorMessages` gets set when the server responds with an error. So if validation fails because the server responsed to the *previous* value with an error, it won't let us try again.

(Which will become relevant if the API ever fails for `url` with any extra validation.)